### DISCUSSION TITLE
Joining Cycloalkanes

### DISCUSSION THREAD

**Discussion by shyamramesh** [rdkit/rdkit#6672] [Ideas]
How to join cycloalkanes at multiple places to create honeycomb like structures. How do I find out bonds that can be joined to form such a structure? in python

**Comment by dehaenw** [comment]
this is a kind of vague question but one way to do this is via a rxnSMARTS that closes six membered rings and some way to prevent the formation of complicated bicycles. below is one way to use this:
```python
from rdkit import Chem
from rdkit.Chem import AllChem,Draw,rdMolDescriptors
import random

bh = lambda mol: rdMolDescriptors.CalcNumBridgeheadAtoms(mol)
RXN = AllChem.ReactionFromSmarts('[C;H2,H3:0][C;H2,H3:1][C:2][C:3][C:4][C:5][C;H2,H3:6][C:7][C:8][C;H2,H3:9]>>[*:0]1[*:1]2[*:2][*:3][*:4][*:5][*:6]2[*:7][*:8][*:9]1') #3+3

def cyclize_alkane(mol, max_steps=100, seed=0):
    cyclized = Chem.Mol(mol) #copy mol
    random.seed(seed)
    step = 0
    products = [(cyclized,)]
    while step < max_steps and len((products)) > 0:
        cyclized = random.choice(products)[0]
        Chem.SanitizeMol(cyclized)
        products = RXN.RunReactants((cyclized,))
        products = [p for p in products if bh(p[0]) == 0]
        step += 1
    return cyclized

mol = Chem.MolFromSmiles(50 * "C") #C50 alkane
honeycombs = [cyclize_alkane(mol, seed = i) for i in range(50)]
d = Draw.MolsToGridImage(honeycombs, molsPerRow = 10)
display(d)
```

the used rxnsmarts corresponds to the below pattern:

![image](https://github.com/rdkit/rdkit/assets/66372095/39f363f8-ea28-429b-aec9-c4369cc9fb7d)

the output of the above codeblock is the following image with random honeycomb like structures:
![image](https://github.com/rdkit/rdkit/assets/66372095/7db7dcbf-9cae-4f09-aaae-73d868233b66)

**Reply by shyamramesh** [reply]
Thanks for taking time to help. Apologies for being vague. My question is how to bridge below molecules to create a ring structure.
molecule 1 - C12C(C3CC2)(CCC3)CCC1 
molecule2 - C1CCCC1

**Reply by dehaenw** [reply]
you can fuse them using rxn SMARTS `[C;H2,H1:0][C;H2,H1:1].[r:2][C;H2,H1;r:3][C;H2,H1,r:4][*;r:5]>>[*:0](-[*:2])[*:1](-[*:5]).[*:3][*:4]`
essentially, this fuses both cycloalkanes at some two carbon juncture.
adapting that to your example molecule and the codeblock above we get:
```python
from rdkit import Chem
from rdkit.Chem import AllChem,Draw,rdMolDescriptors,rdDepictor
import random

bh = lambda mol: rdMolDescriptors.CalcNumBridgeheadAtoms(mol)
RXN = AllChem.ReactionFromSmarts('[C;H2,H1:0][C;H2,H1:1].[r:2][C;H2,H1;r:3][C;H2,H1,r:4][*;r:5]>>[*:0](-[*:2])[*:1](-[*:5]).[*:3][*:4]')

def cyclize_alkane(mol1,mol2,max_steps=100, seed=0):
    cyclized = Chem.Mol(mol1) #copy mol
    mol2 = Chem.Mol(mol2) #copy mol
    random.seed(seed)
    step = 0
    products = [(cyclized,mol2)]
    while step < max_steps and len((products)) > 0:
        cyclized = random.choice(products)[0]
        Chem.SanitizeMol(cyclized)
        products = RXN.RunReactants((cyclized,mol2))
        products = [p for p in products if bh(p[0]) == 0]
        step += 1
    return cyclized

mol = Chem.MolFromSmiles("C12C(C3CC2)(CCC3)CCC1") #molecule1
mol2 = Chem.MolFromSmiles("C1CCCC1")
honeycombs = [cyclize_alkane(mol,mol2, max_steps=15, seed = i) for i in range(25)]
d = Draw.MolsToGridImage(honeycombs, molsPerRow = 5)
display(d)
```
giving some "honeycomb" like structures but with five membered rings insetad
